### ISSUE TITLE
How to Improve Accuracy in this Work？

### ISSUE THREAD

**Issue by zhenzhongle** [Daniil-Osokin/lightweight-human-pose-estimation.pytorch#169]
I used many other network models to replace the original Mobile net, such as ghostnet, Octmobile net, and so on.  But their losses fall slowly and remain constant for a long time. Is there any way to improve accuracy ?

**Comment by zhenzhongle** [comment]
Thank you for your reply.  
 My idea is based on your work ( checkpoint _ iter _ 370000 reaches map = 40 % ).  Improved Mobile net to over 40 % or make the model lighter, but I changed several pre-training models for nearly two years and found that the effect was not as good as mobilenet _ sgd _ 68.848.pth。I sincerely wish you can provide some idea.  
Finally wish you good health, work smoothly !

**Comment by Daniil-Osokin** [comment]
Oh, get it. You can always add the attention (e.g. ["Squeeze-and-Excitation Networks"](https://arxiv.org/pdf/1709.01507.pdf)), it will likely boost the accuracy :smiley:. One more way to improve the accuracy is to increase network output resolution (and retrain): now it downsamples input 8 times, the most of other models downsamples input just 4 times. I believe the crucial thing to add is multi-scale supervision (e.g. ["Multi-Scale Structure-Aware Network for Human Pose Estimation"](https://openaccess.thecvf.com/content_ECCV_2018/papers/Lipeng_Ke_Multi-Scale_Structure-Aware_Network_ECCV_2018_paper.pdf), ["HigherHRNet: Scale-Aware Representation Learning for Bottom-Up Human Pose Estimation"](https://arxiv.org/pdf/1908.10357.pdf)). One more trick is voting maps for improved keypoints locations: each output pixel in a map(s) predicts shift to the nearest kepoint(s), or kind of (e.g. ["PifPaf: Composite Fields for Human Pose Estimation"](https://arxiv.org/pdf/1903.06593.pdf), ["Distribution-Aware Coordinate Representation for Human Pose Estimation"](https://arxiv.org/pdf/1910.06278.pdf)). There are also some minor tricks, such as deformable convolutions, hard examples mining, and so on. I believe EfficientNet backbone (with pre-trained weights) will improve the accuracy if you just use it instead of MobileNet. This is about accuracy improvement. Improve inference speed should be relatively easy (but tricks differ from particular inference device), at least distillation or pruning will work.

**Comment by zhenzhongle** [comment]
Thank you for your proposal, I will seriously reflectionand try, thank you very very much!
You are a responsible and friendly ‘blogger’

**Comment by Daniil-Osokin** [comment]
Ok, closing it.

**Comment by sulei1998** [comment]
> 哦，明白了。您可以随时添加注意力（例如[“挤压和激励网络”](https://arxiv.org/pdf/1709.01507.pdf)），这可能会提高准确性😃.提高准确性的另一种方法是提高网络输出分辨率（并重新训练）：现在它将输入下采样 8 倍，大多数其他模型仅对输入进行下采样 4 倍。我认为要添加的关键是多尺度监督（例如[“用于人体姿势估计的多尺度结构感知网络”，“](https://openaccess.thecvf.com/content_ECCV_2018/papers/Lipeng_Ke_Multi-Scale_Structure-Aware_Network_ECCV_2018_paper.pdf)[HigherHRNet：用于自下而上的人体姿态估计的尺度感知表示学习”](https://arxiv.org/pdf/1908.10357.pdf)）。另一个技巧是投票地图以改进关键点位置：地图中的每个输出像素预测移动到最近的kepoint或类型（例如[“PifPaf：用于人体姿势估计的复合场”，“](https://arxiv.org/pdf/1903.06593.pdf)[用于人体姿势估计的分布感知坐标表示”](https://arxiv.org/pdf/1910.06278.pdf)）。还有一些小技巧，例如可变形卷积，硬示例挖掘等。我相信EfficientNet骨干网（带有预先训练的权重）将提高准确性，如果你只是使用它而不是MobileNet。这是关于准确性的提高。提高推理速度应该相对容易（但技巧与特定的推理设备不同），至少蒸馏或修剪会起作用。

Hello, what you mean by adding attention or multi-scale supervision is to modify the network structure of the backone (MobileNet V1)?
